internal/services/hdinsight/hdinsight_spark_cluster_resource.go

```diff
@@ -272,6 +274,10 @@ func resourceHDInsightSparkClusterCreate(d *pluginsdk.ResourceData, meta interfa
 		}
 	}
 
+	if v, ok := d.GetOk("availability_zones"); ok {
+		payload.Zones = pointer.To(expandHDInsightAvailabilityZones(v))
```

Comment:
There are functions available to expand and flatten zones
```suggestion
		payload.Zones = pointer.To(zones.ExpandUntyped(d.Get("zones").(*schema.Set).List()))
```

Reply:
Changed.